pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -2043,6 +2129,29 @@ func unpublishCertObjConfig(getconfigCtx *getconfigContext, uuidStr string) {
 	pub.Unpublish(key)
 }
 
+// parseCpherBlock : handle cipher information received from controller
+// consolidate both the zconfig.CipherContext and zconfig.CipherBlock to types.CipherBlock.
+// The interested agents will get the whole information
+func parseCipherBlock(ctx *getconfigContext, config *zconfig.CipherBlock) (*types.CipherBlock, bool) {
```

Comment:
Return error type for the two different errors and have all the callers pass such errors back to the controller.

Reply:
sure, will do.